src/components/tools/hooks/use-measure-text.ts

```diff
@@ -1,15 +1,19 @@
 import { useCallback, useRef } from "react";
 
-export const useMeasureText = (font: string) => {
+const defaultFont = "italic 14px Lato, sans-serif";
+
+export const measureText =
+  (text:string, canvas:HTMLCanvasElement, cache:Record<string, number> = {}, font:string = defaultFont) => {
+  const context = canvas.getContext("2d");
+  context && font && (context.font = font);
+  cache[text] = cache[text] || (context ? Math.ceil(10 * context.measureText(text).width) / 10 : 0);
+  return cache[text];
+};
+
+export const useMeasureText = (font: string = defaultFont) => {
   const cache = useRef<Record<string, number>>({});
   const canvas = useRef(document.createElement("canvas"));
```

Comment:
> But isn't `useRef()` only evaluated once, on initialization, anyway? Or is the argument evaluated every render but only assigned on initialization?

The latter -- the argument is evaluated on every render but `useRef()` continues to return the first instance. `useMemo()` doesn't have the same behavior because of the indirection of the function -- it only executes the function when it needs to.